### PR TITLE
renamed cur_cat => cur_cat_idx and added some comments

### PR DESCRIPTION
**Pull request by zyxue**
just for clarity and trying to make it easier to follow the code for those who aren't familiar with the code base.

### CODE REVIEW

**Review by jameslamb — DISMISSED** [review]

File: src/io/bin.cpp [hunk 1]
```diff
@@ -466,21 +465,22 @@ namespace LightGBM {
         bin_2_categorical_.push_back(-1);
         categorical_2_bin_[-1] = 0;
         cnt_in_bin.push_back(0);
+        size_t cur_cat_idx = 0; // index of current category.
```
[the remark this answers is not in the document]
Comment (zyxue) [reply src/io/bin.cpp:468]: the idea is to just initialize a new variable closer to where it's first used, which helps with clarity.

[the remark this answers is not in the document]
Comment (zyxue) [reply src/io/bin.cpp:468]: I can remove this change if it's undesired.

[the remark this answers is not in the document]
Comment (jameslamb) [reply src/io/bin.cpp:468]: yes please revert that change, to keep this focused on only one purpose (renaming)

**Review by jameslamb — COMMENTED** [review]
Thanks for the changes.

This looks ok to me, but I'll just leave a "comment" review and let @guolinke or @shiyu1994 check that this new name correctly reflects what the code is doing.

Until then... can you please merge in the latest state of `master`? We just merged a fix for the failing CI jobs (#5510). Sorry for the inconvenience.

**Review by guolinke — APPROVED** [review]
LGTM

**Comment by jameslamb** [comment]
@zyxue please push a new commit addressing the linting errors ([build link](https://github.com/microsoft/LightGBM/actions/runs/3206906845/jobs/5241214758)).

```text
src/io/bin.cpp:355:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/io/bin.cpp:392:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/io/bin.cpp:454:  At least two spaces is best between code and comments  [whitespace/comments] [2]
```

**Review by jameslamb — APPROVED** [review]
Looks good, thanks! We will merge this when it builds.

**Pull request merged**